ruma-api-macros/src/api/request.rs

```diff
@@ -592,6 +592,214 @@ impl Request {
                             }
                         }
     }
+
+    /// The first item in the tuple generates code for the request path from
+    /// the `Metadata` and `Request` structs. The second item in the returned tuple
+    /// is the code to generate a Request struct field created from any segments
+    /// of the path that start with ":".
+    ///
+    /// The first `TokenStream` returned is the constructed url path. The second `TokenStream` is
+    /// used for implementing `TryFrom<http::Request<Vec<u8>>>`, from path strings deserialized to Ruma
+    /// types.
```

Comment:
Please re-wrap these (doc) comments so they don't go over 100 characters. Merging shorter lines would also be nice.